srcs/commands/Join.cpp

```diff
@@ -0,0 +1,94 @@
+#include "../../includes/Commands.hpp"
+#include "../../includes/Server.hpp"
+
+static std::string getChannelName(std::string channel)
+{
+	std::size_t found = channel.find('#');
+	if (found != std::string::npos)
+		return (channel);
+	else
+		return ("#" + channel);
+}
+
+static int joinExistingServerWithoutKey(std::map<std::string, Channel*> &channels, std::string channelName, Client *Client)
+{
+	std::map<std::string, Channel*>::iterator it;
+	for (it=channels.begin(); it!=channels.end(); it++)
+	{
+		if (it->first == channelName)
+		{
+			if (it->second->getChannelKey().empty() == false)
+			{
+				send(Client->getClientFd(), ERR_BADCHANNELKEY(Client->getHostName(), channelName).c_str(), ERR_BADCHANNELKEY(Client->getHostName(), channelName).length(), 0);
+				return (-1);
+			}
+			it->second->addToChannel(*Client);
+			return (0);
+		}
+	}
+	channels[channelName] = new Channel(channelName);
+	channels[channelName]->addToChannel(*Client);
+	return (0);
+}
+
+static int joinExistingServerWithKey(std::map<std::string, Channel *> &channels, std::string channelName, std::string key, Client *Client)
+{
+	std::map<std::string, Channel*>::iterator it;
+	for (it=channels.begin(); it!=channels.end(); it++)
+	{
+		if (it->first == channelName)
+		{
+			if (it->second->getChannelKey() == key)
+			{
+				it->second->addToChannel(*Client);
+				return (0);
+			}
+			else
+			{
+				send(Client->getClientFd(), ERR_BADCHANNELKEY(Client->getHostName(), channelName).c_str(), ERR_BADCHANNELKEY(Client->getHostName(), channelName).length(), 0);
+				return (-1);
+			}
+		}
+	}
+	channels[channelName] = new Channel(channelName);
+	channels[channelName]->addToChannel(*Client);
+	channels[channelName]->setChannelKey(key);
+	return (0);
+}
+
+int cmdJoin(Message &msg, Client *Client, std::map<std::string, Channel*> &channels)
```

Comment:
Do we need an error in case of channel name is already in use?

Reply:
It is only partly ready so those will be added later. I am pushing it so Juan can test quit commands with one channel :)